util/reconciler/conditions.go

```diff
@@ -31,13 +31,12 @@ func HasConditionSeverity(from conditions.Getter, typ clusterv1.ConditionType, s
 }
 
 func RecordDecayingCondition(to conditions.Setter, typ clusterv1.ConditionType, reason, message string, timeout time.Duration) bool {
-	conditions.MarkFalse(to, typ, reason, clusterv1.ConditionSeverityWarning, "%s", message)
-
 	if HasStaleCondition(to, typ, timeout) {
 		conditions.MarkFalse(to, typ, reason, clusterv1.ConditionSeverityError, "%s", message)
 		return true
 	}
 
+	conditions.MarkFalse(to, typ, reason, clusterv1.ConditionSeverityWarning, "%s", message)
```

Comment:
won't this set the severity to warning for a stale condition instead of error?

Reply:
`HasStaleCondition()` method returns true for stale condition and sets the severity to error and returns. SeverityWarning is only added if the condition doesn't exist or if its not stale.